### PR TITLE
Prepare for v0.4.0

### PR DESCRIPTION
**Pull request by robbkidd**

### CODE REVIEW

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
image
 Nice.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # honeycomb-opentelemetry-java changelog
 
+## [0.4.0] - 2021-08-12
+
+### Changes
+
+- Disabled metrics from the agent by default (#92) (see #96 for the plan to add them back in) | [@JamieDanielson](https://github.com/JamieDanielson)
+- Make API Key and Dataset optional (#89) | [@vreynolds](https://github.com/vreynolds)
+- Add deterministic sampler tests (#73) (7 weeks ago) [Mike Goldsmith]
```
Comment (vreynolds) [inline CHANGELOG.md:9]:
```suggestion
- Add deterministic sampler tests (#73) | [@MikeGoldsmith](https://github.com/MikeGoldsmith)
```

**Review by vreynolds — APPROVED** [review]
smol fix for Mike's name in the changelog

File: CHANGELOG.md (hunk 1 above)
Comment (robbkidd) [reply CHANGELOG.md]: How could I miss, Mike! 🤦🏻

**Pull request merged**